File: src/601-700/686.cpp
#include <iostream>
#include <cmath>
#include <cassert>
#include <cstdint>

using namespace std;

int main() {
    long double const log10_2 = log10l(2.0L);

    int64_t n = 6;
    int cnt = 0;
    while(cnt < 678910) {
        n++;
        int64_t len = (int64_t)ceil(log10_2 * n);
        int64_t offset = len - 3;
        int prefix = (int)floor(powl(10.0L, log10_2 * n - offset));
        if (prefix == 123) {
            cnt++;
            // cout << cnt << ' ' << n << ' ' << prefix << ' ' << len << '\n';
        }
    }
    cout << n << '\n';
}

File: src/601-700/686.py
n = 45
cnt = 0
k = 0
pow2 = 1

while cnt < n:
    k += 1
    pow2 *= 2
    prefix_pow2 = pow2
    while prefix_pow2 >= 1000:
        prefix_pow2 //= 10
    if prefix_pow2 == 123:
        # print(k, 2**k)
        cnt += 1

print(k)